runtime/src/bank/pyth_accumulator_tests.rs

```diff
@@ -0,0 +1,890 @@
+                price_feed_info_lamports,
+                price_feed_info_data,
+                price_feed_info_owner,
+                false,
+                Epoch::default(),
+            );
+
+            let mut price_account = PriceAccount::initialize(&price_feed_info, 0).unwrap();
+            if !generate_buffers {
+                price_account.flags.insert(
+                    PriceAccountFlags::ACCUMULATOR_V2 | PriceAccountFlags::MESSAGE_BUFFER_CLEARED,
+                );
+            }
+
+            vec![
+                price_account
+                    .as_price_feed_message(&price_feed_key.to_bytes().into())
+                    .to_bytes(),
+                price_account
+                    .as_twap_message(&price_feed_key.to_bytes().into())
+                    .to_bytes(),
+            ]
+        };
+
+        bank.store_account(&price_feed_key, &price_feed_account);
+
+        if generate_buffers {
+            let message_buffer_bytes = create_message_buffer_bytes(messages.clone());
+
+            let mut seed = vec![1; 32];
+            seed[..seeds.len()].copy_from_slice(seeds);
+            // Create a Message account.
+            let price_message_key = keypair_from_seed(&seed).unwrap();
+            let mut price_message_account = bank
+                .get_account(&price_message_key.pubkey())
+                .unwrap_or_default();
+
+            price_message_account.set_lamports(1_000_000_000);
+            price_message_account
+                .set_owner(Pubkey::new_from_array(pythnet_sdk::MESSAGE_BUFFER_PID));
+            price_message_account.set_data(message_buffer_bytes);
+
+            // Store Message account so the accumulator sysvar updater can find it.
+            bank.store_account(&price_message_key.pubkey(), &price_message_account);
+        }
+
+        (price_feed_key, messages)
+    };
+
+    assert!(bank
+        .feature_set
+        .is_active(&feature_set::enable_accumulator_sysvar::id()));
+    assert!(bank
+        .feature_set
+        .is_active(&feature_set::move_accumulator_to_end_of_block::id()));
+    assert!(bank
+        .feature_set
+        .is_active(&feature_set::undo_move_accumulator_to_end_of_block::id()));
+    assert!(bank
+        .feature_set
+        .is_active(&feature_set::redo_move_accumulator_to_end_of_block::id()));
+
+    let prices_with_messages = [
+        generate_price(b"seeds_1", generate_buffers[0]),
+        generate_price(b"seeds_2", generate_buffers[1]),
+        generate_price(b"seeds_3", generate_buffers[2]),
+        generate_price(b"seeds_4", generate_buffers[3]),
+    ];
+
+    bank = new_from_parent(&Arc::new(bank)); // Advance slot 1.
+    bank = new_from_parent(&Arc::new(bank)); // Advance slot 2.
+
+    let messages = prices_with_messages
+        .iter()
+        .map(|(_, messages)| messages)
+        .flatten()
+        .map(|message| &message[..])
+        .sorted_unstable()
+        .dedup()
+        .collect::<Vec<_>>();
+
```

Comment:
nitpick: I'd assert the length of `messages` here, to confirm they haven't been deduplicated

Reply:
Added.